container-search/src/test/java/ai/vespa/search/llm/RAGSearcherWithOpenAITest.java

```diff
@@ -0,0 +1,115 @@
+
+    private static final String DOC1_TITLE = "Exploring the Delightful Qualities of Ducks";
+    private static final String DOC1_CONTENT = "Ducks, with their gentle quacks and adorable waddling walks, possess a unique " +
+            "charm that sets them apart as extraordinary pets.";
+    private static final String DOC2_TITLE = "Why Cats Reign Supreme";
+    private static final String DOC2_CONTENT = "Cats bring an enchanting allure to households with their independent " +
+            "companionship, playful nature, natural hunting abilities, low-maintenance grooming, and the " +
+            "emotional support they offer.";
+
+
+    public static class MockSearchResults extends Searcher {
+
+        @Override
+        public Result search(Query query, Execution execution) {
+            Hit hit1 = new Hit("1");
+            hit1.setField("title", DOC1_TITLE);
+            hit1.setField("content", DOC1_CONTENT);
+            hit1.setField("matchfeatures", "...");
+
+            Hit hit2 = new Hit("2");
+            hit2.setField("title", DOC2_TITLE);
+            hit2.setField("content", DOC2_CONTENT);
+            hit2.setField("summaryfeatures", "...");
+            hit2.setField("rankfeatures", "...");
+
+            Result r = new Result(query);
+            r.hits().add(hit1);
+            r.hits().add(hit2);
+            return r;
+        }
+    }
+    
+    @Test
+    public void testStreaming() {
+        var llmConfig = new LlmClientConfig.Builder()
+                .apiKeySecretName("openai")
+                .maxTokens(100)
+                .build();
+        var openai = new OpenAI(llmConfig, new MockSecrets());
+
+        var searcherConfig = new LlmSearcherConfig.Builder().stream(true).build();
+        ComponentRegistry<LanguageModel> models = new ComponentRegistry<>();
+        models.register(ComponentId.fromString("openai"), openai);
+        models.freeze();
+        var searcher = new RAGSearcher(searcherConfig, models);
+
+        var chain = new Chain<>(searcher, new MockSearchResults());
+        var execution = new Execution(chain, Execution.Context.createContextStub());
+        var queryParams = Map.of(
+                "query", "why are ducks better than cats?",
+                "llm.fields", "title,content",
+                "llm.prompt", "{context}\nGiven these documents, answer this query as concisely as possible: @query"
+        );
+        var query = new Query("?" + LLMSearcherTest.toUrlParams(queryParams));
+
+        var result = execution.search(query);
+        var stream = (EventStream) result.hits().get(0);
+        // assert that stream is not complete - (if completeAsync had been blocking, it would be complete)
+        // For this assertion to fail, OpenAI-completion must have been completed in less time than the execution time
+        // of returning the search. Consider this more robust than asserting on timings.
+        assert !stream.incoming().isComplete();
```

Comment:
Better to use assertFalse, more consistent with other tests.
```suggestion
       assertFalse(stream.incoming().isComplete()))
```